File: 2/C.bmo.cpp
#include <math.h>
#include <stdio.h>

using namespace std;

int main()
{

    int n;
    int k;
    int summer;
    int tmp;
    scanf("%d %d", &n, &k);

    float q = 1.0/k;
    float q30 = pow(q, 30);
    float f = (1-q30) / (1-q);

    if (k > n)
    {
        printf("%d", n);
    }
    else
    {
        for (int i = n - n/k; i < 10000000000; ++i)
        {
            int a = i * f;
            if (a >= n)
            {
                summer = i;
                tmp = i;

                while (tmp >= k)
                {
                    tmp = tmp/k;
                    summer += tmp;
                }

                if (summer >= n)
                {
                    printf("%d", i);
                    break;
                }
            }
        }
   }

   return 0;
}

File: 2/C.bmo.py
import math
n, k = map(int, input().split())
q = 1/k
q20 = q**40
f = (1-q20) / (1-q)
if k > n:
    print (1)
else:
    for i in range(int(n/2),10000000000):
        a = i * f
        if a > n:
            summer = int(i)
            tmp = summer
            while tmp >= k:
                a = int(tmp / k)
                summer += a
                tmp = a
            if summer >= n:
                print(i)
                break